pkg/types/git_project.go

```diff
@@ -24,8 +26,10 @@ func (gp *GitProject) UnmarshalYAML(unmarshal func(interface{}) error) error {
 
 func ValidateGitProject(sl validator.StructLevel) {
 	gp := sl.Current().Interface().(GitProject)
-	if strings.Index(gp.SubDir, "/") != -1 {
-		sl.ReportError(gp.SubDir, "subDir", "SubDir", "/ is not allowed in subDir", "")
+	for _, e := range strings.Split(gp.SubDir, string(os.PathSeparator)) {
```

Comment:
The subPath in question here is not os-dependent so it must split by `/`.
While writing this I realised that this validation should also check for `\` to not be existent in the path and that there might also be many other places where such checks would be required.

This sounds like it should be moved into a completely new PR with only path related fixes. Do you want me to take over that part or do you also want to do that? I think I will find some time tomorrow to work on Kluctl topics again.

Reply:
I can have a look. Then we probably want to match it some regex pattern. `[-_\.a-zA-Z0-9/]+` will have to check whether at least part is fully covered by `\w` in golang. Though regex will take more cpu cycles however on dedicated fuction it could be reasonable.